### PR TITLE
Harden UI thread safety

### PR DESCRIPTION
**Pull request by PawelLipski**
Follow-up to #1211

### CODE REVIEW

File: branchLayout/impl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutFileWriter.java [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
 import com.virtuslab.branchlayout.api.IBranchLayoutEntry;
 import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
 
 @ExtensionMethod(BranchLayoutFileUtils.class)
 @CustomLog
 @RequiredArgsConstructor
 public class BranchLayoutFileWriter implements IBranchLayoutWriter {
 
   // TODO (#692): wrap write operation in WriteAction
```
Comment (PawelLipski) [inline branchLayout/impl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutFileWriter.java:26]: Unrel... @mkondratek can it be removed? as #692 is closed 🤔

File: src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java [hunk 2]
```diff
@@ -75,9 +79,7 @@ public void check(JavaMethod method, ConditionEvents events) {
 
             method.getCallsFromSelf().forEach(call -> {
               AccessTarget accessTarget = call.getTarget();
-              if (call.isDeclaredInLambda()) {
-                // ignore
-              } else if (accessTarget.isAnnotatedWith(UIThreadUnsafe.class)) {
+              if (accessTarget.isAnnotatedWith(UIThreadUnsafe.class)) {
```
Comment (mkondratek) [inline src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java:82]: 👍

File: branchLayout/impl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutFileWriter.java (hunk 1 above)
Comment (mkondratek) [reply branchLayout/impl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutFileWriter.java:26]: yes, I think we can remvoe it

**Review by mkondratek — APPROVED** [review]
lgtm

File: frontend/file/build.gradle.kts [hunk 3]
```diff
@@ -41,9 +41,6 @@ sourceSets["main"].java { srcDir(additionalSourceDirs) }
 
 val generateMacheteParser =
   tasks.withType<GenerateParserTask> {
-    // See https://github.com/JetBrains/gradle-grammar-kit-plugin/issues/89
```
Comment (PawelLipski) [inline frontend/file/build.gradle.kts:44]: Unrel, piggyback 😅

**Review by mkondratek — APPROVED** [review]

**Pull request merged**